### PR TITLE
refactor(runtime): remove `<RenderHook>`

### PR DESCRIPTION
**Pull request by agurtovoy**
Rewrite element tree rendering using "unified" controls interface.

### CODE REVIEW

**Comment by joshuawootonn** [comment]
👁️ 👁️

File: packages/runtime/src/runtimes/react/controls.tsx [hunk 1]
```diff
@@ -1,164 +1,41 @@
...
-  )
-  const props = element.props as Record<string, any>
-  const documentKey = useDocumentKey()
+  const all = ReactPage.getComponentPropControllerDescriptors(store.getState(), elementType) ?? {}
+  return useRef(partitionRecord(all, isLegacyDescriptor)).current
+}
 
-  const propControllers = useSelector(state => {
-    if (documentKey == null) return null
+export function ResolveProps({ element, children: renderComponent }: PropsValueProps): JSX.Element {
```
Comment (agurtovoy) [inline packages/runtime/src/runtimes/react/controls.tsx:28]: Renamed `PropsValue` to `ResolveProps`.

File: packages/runtime/src/runtimes/react/controls.tsx [hunk 2]
```diff
@@ -1,164 +1,41 @@
...
+  const [legacyDescriptors, definitions] = useControlDefs(element.type)
 
-    return ReactPage.getPropControllers(state, documentKey, element.key)
-  })
+  const resolvedProps = useResolvedProps(definitions, element.props, element.key)
```
Comment (agurtovoy) [inline packages/runtime/src/runtimes/react/controls.tsx:31]: `useResolvedProps` implements prop resolution for all of the new controls.

File: packages/runtime/src/runtimes/react/controls.tsx [hunk 3]
```diff
@@ -1,164 +1,41 @@
...
 
-  return Object.entries(propControllerDescriptorsRef.current).reduceRight(
+  const renderFn = Object.entries(legacyDescriptors).reduceRight(
```
Comment (agurtovoy) [inline packages/runtime/src/runtimes/react/controls.tsx:33]: Properties coming from legacy descriptors aka prop controllers are still resolved as before, outside of the unified controls framework.

**Pull request merged**